book/python/using-connectors.md

```diff
@@ -0,0 +1,191 @@
+
+Notice how we don't couple the specifics of the medium used by the connector and instead focus on how we interpret incoming messages in terms of their serialization. This allows applications to evolve and mix their infrastucture over time.
+
+## Running Connector Scripts
+
+To run an application that makes use of these connectors we'll need to understand how the connector scripts work. You can find a number of the included connectors [here](https://github.com/WallarooLabs/wallaroo/tree/{{book.wallaroo_version}}/connectors/). We'll look at `udp_source` and `udp_sink` as used in the example introduced above.
+
+These are self-contained python scripts. Some of them may expect a related python library to be installed and available for import but our example uses the UDP facilities provider for by Python's standard library. Be sure that your python path is set up to include the wallaroo library.
+
+If we run one of these scripts we'll see there are some required arguments listed. The first one should be familiar. `--application-module` is the same as the one seen when running a Wallaroo worker (if you need a review of those command line parameters check the [Running Wallaroo](/book/running-wallaroo/running-wallaroo.md) section). Providing this argument allows the script to autodetect most of your settings by reading the same application description that your wallaroo app runs with.
+
+The application module is not quite enough for the script to know what settings to use when connecting to wallaroo. Since your application may have more than one connector, the `--conncetor` argument is required for disambiguating which connector in your application setup this script is running as. Once you pass this in, we can autodetect the most common settings.
+
+Many connector scripts will have their own additional arguments, some may be required and others may be optional. In this case we have two, the host and port. These argument names prefix the connector name to ensure that conflicting argument names do not cause problems (should be able to pass all your wallaroo arguments in without conflict for example).
+
+To illustrate this, let's look at what the `udp_source` command line parameters might look like:
+
+```bash
+./wallaroo/connectors/udp_source --application-module celsius --connector celsius_feed --celsius_feed-host 127.0.0.1 --celsius_feed-port 6789
+```
+
+In this example, we are running a UDP source for our application which listens to a UDP socket on 127.0.0.1 port 6789 and forwards the data in each packet to wallaroo as a message.
+
+Since the included connector scripts are self-contained, feel free to copy it over to your application tree to use them directly and make path management a little easier.
+
+## Starting With Prebuilt Connectors
+
+We have a number of [prebuild connectors](https://github.com/WallarooLabs/wallaroo/tree/{{book.wallaroo_version}}/connectors/). Here is a short description of each one.
```

Comment:
WDYT about talking about how these are more starting points than prebuilt connectors?

Reply:
I try to speak to that above but I'll repeat the note about being able to copy these into your project and customize them.